homework/index.js

```diff
@@ -30,13 +30,100 @@
     return elem;
   }
 
+  function removeChildElements(parent) {
+    while (parent.firstChild) {
+      parent.removeChild(parent.firstChild);
+    }
+  }
+
+  function displayError(error, nodeToDisplay) {
+    createAndAppend('p', nodeToDisplay, { text: error.message, class: 'alert-error' });
+  }
+  function showContributors(contributors) {
+    const contributorsContainer = document.getElementById('contributors');
+    if (contributors) {
+      const ul = createAndAppend('ul', contributorsContainer);
+      for (const contributor of contributors) {
+        const li = createAndAppend('li', ul);
+        createAndAppend('img', li, { src: contributor.avatar_url, class: 'avatar' });
+        createAndAppend('a', li, {
+          text: contributor.login,
+          href: contributor.html_url,
+          class: 'contributor',
+          target: '_blank',
+        });
+        createAndAppend('div', li, {
+          text: contributor.contributions,
+          class: 'contribution-count',
+        });
+      }
+    } else {
+      createAndAppend('p', contributorsContainer, {
+        text: 'No Contributors',
+        class: 'no-contributor',
+      });
+    }
+  }
+
+  function showRepositoryDetails(data) {
+    const repositoryContainer = document.getElementById('repository');
+    removeChildElements(repositoryContainer);
+    const selected = document.getElementById('repositorySelect').selectedIndex;
+    const repository = data[selected];
+    createAndAppend('img', repositoryContainer, {
+      src: 'hyf.png',
+      alt: 'Hack Your Future Logo',
+      class: 'logo',
+    });
+    const h2 = createAndAppend('h2', repositoryContainer, { id: 'repositoryName' });
+    createAndAppend('a', h2, {
+      text: repository.name,
+      href: repository.html_url,
+      target: '_blank',
+    });
+    createAndAppend('p', repositoryContainer, { text: repository.description });
+
+    const contributorsContainer = document.getElementById('contributors');
+
+    fetchJSON(repository.contributors_url, (error, contributors) => {
+      removeChildElements(contributorsContainer);
+      if (error) {
+        displayError(error, contributorsContainer);
+      } else {
+        showContributors(contributors);
+      }
+    });
+  }
+
+  function createHeader(data) {
+    const header = createAndAppend('header', root, { id: 'header' });
+    createAndAppend('h1', header, { text: 'Hack Your Future Repositories' });
+    const nav = createAndAppend('nav', root, { id: 'navigation' });
+    const select = createAndAppend('select', nav, { id: 'repositorySelect' });
```

Comment:
Hi Fatih - if we go back to our first JavaScript lessons, you would remember that it is a good practice to use kebab-casing rather than camelCasing for class names and ID attribute values in HTML elements. So instead of having `{ id: 'repositorySelect' }`, you should try for `{ id: 'repository-select' }`.